cmd/podman/trust.go

```diff
@@ -0,0 +1,557 @@
+				if strings.Contains(uid, "<") && strings.Contains(uid, ">") {
+					parseduid = strings.SplitN(strings.SplitAfterN(uid, "<", 2)[1], ">", 2)[0]
+				}
+				parseduids = append(parseduids, parseduid)
+			}
+		}
+		return strings.Join(parseduids, ",")
+	}
+	return ""
+}
+
+func createKeyTmpFile(content []byte) (string, error) {
+	tmpfile, err := ioutil.TempFile("/run/", "")
+	if err != nil {
+		return "", err
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		return "", err
+
+	}
+	if err := tmpfile.Close(); err != nil {
+		return "", err
+	}
+	return tmpfile.Name(), nil
+}
+
+var typeDescription = map[string]string{"insecureAcceptAnything": "accept", "signedBy": "signed", "reject": "reject"}
+
+func trustTypeDescription(t string) string {
+	ele, exist := typeDescription[t]
+	if !exist {
+		logrus.Warnf("invalid trust type %s", t)
+	}
+	return ele
+}
+
+func sortPolicyJSONKey(m map[string]map[string]interface{}) []string {
+	keys := make([]string, len(m))
+	i := 0
+	for k := range m {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+// defaultPolicyPath returns a path to the default policy of the system.
+func defaultPolicyPath(sys *types.SystemContext) string {
+	systemDefaultPolicyPath := "/etc/containers/policy.json"
+	if sys != nil {
+		if sys.SignaturePolicyPath != "" {
+			return sys.SignaturePolicyPath
+		}
+		if sys.RootForImplicitAbsolutePaths != "" {
+			return filepath.Join(sys.RootForImplicitAbsolutePaths, systemDefaultPolicyPath)
+		}
+	}
+	return systemDefaultPolicyPath
+}
+
+func getPolicyJSON(policyContentStruct PolicyContent, systemRegistriesDirPath string) (map[string]map[string]interface{}, error) {
+	registryConfigs, err := loadAndMergeConfig(systemRegistriesDirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	policyJSON := make(map[string]map[string]interface{})
+	if len(policyContentStruct.Default) > 0 {
+		policyJSON["* default"] = make(map[string]interface{})
+		policyJSON["* default"]["type"] = policyContentStruct.Default[0].Type
+	}
+	for transname, transval := range policyContentStruct.Transports {
+		for repo, repoval := range transval {
+			policyJSON[repo] = make(map[string]interface{})
+			policyJSON[repo]["type"] = repoval[0].Type
+			policyJSON[repo]["transport"] = transname
+			for _, repoele := range repoval {
+				keyarr := []string{}
+				if len(repoele.KeyPath) > 0 {
+					keyarr = append(keyarr, repoele.KeyPath)
+				}
+				if len(repoele.KeyData) > 0 {
+					keyarr = append(keyarr, string(repoele.KeyData))
+				}
+				policyJSON[repo]["keys"] = keyarr
+			}
+			policyJSON[repo]["sigstore"] = ""
+			registryNamespace := haveMatchRegistry(repo, registryConfigs)
+			if registryNamespace != nil {
+				policyJSON[repo]["sigstore"] = registryNamespace.SigStore
+			}
+		}
+	}
+	return policyJSON, nil
+}
+
+// registriesDirPath returns a path to registries.d
+func registriesDirPath(sys *types.SystemContext) string {
+	systemRegistriesDirPath := "/etc/containers/registries.d"
```

Comment:
Can we get this path from containers/image?

Reply:
can't find an exported variable, so copied [registriesDirPath(sys *types.SystemContext)](https://github.com/containers/image/blob/9f297120deb84f7da63088cc8e32c541f1b0e963/docker/lookaside.go#L77) from containers/image , the same as defaultPolicyPath.
@mtrmac :thinking:  can I get this path from containers/image?